lib/rack/test/uploaded_file.rb

```diff
@@ -85,19 +87,21 @@ def self.actually_finalize(file)
       private
 
       # Use the StringIO as the tempfile.
-      def initialize_from_stringio(stringio, original_filename)
+      def initialize_from_stringio(stringio)
+        raise(ArgumentError, 'Missing `original_filename` for StringIO object') if original_filename.nil?
+
         @tempfile = stringio
-        @original_filename = original_filename || raise(ArgumentError, 'Missing `original_filename` for StringIO object')
       end
 
-      # Create a tempfile and copy the content from the given path into the tempfile.
+      # Create a tempfile and copy the content from the given path into the tempfile, optionally renaming if
+      # original_filename has been set.
       def initialize_from_file_path(path)
         raise "#{path} file does not exist" unless ::File.exist?(path)
 
-        @original_filename = ::File.basename(path)
-        extension = ::File.extname(@original_filename)
+        @original_filename = original_filename || ::File.basename(path)
+        extension = ::File.extname(original_filename)
 
-        @tempfile = Tempfile.new([::File.basename(@original_filename, extension), extension])
+        @tempfile = Tempfile.new([::File.basename(original_filename, extension), extension])
```

Comment:
No reason to use the accessor here, please switch back to the instance variable.

Reply:
Reverted 👍🏻